### PR TITLE
finish view aws data, upload, and download

### PR DESCRIPTION
**Pull request by zindlerb**

### CODE REVIEW

File: client/js/components/data_page.jsx [hunk 1]
```diff
@@ -1,14 +1,17 @@
...
+var AppStore = require("../store/app_store.js");
 
 var UploadButton = React.createClass({
-  fileHandler: function(event) {
-    console.log(this.files);
-  },
   componentDidMount: function() {
+    var schemaName = this.props.schemaName;
+    function fileHandler(event){
+      AppStore.uploadData(this.files.item(0).name, this.files.item(0), schemaName);
```
Comment (dduugg) [inline client/js/components/data_page.jsx]: consider extracting `this.files.item(0)`

File: client/js/store/app_store.js [hunk 2]
```diff
@@ -24,6 +25,46 @@ var AppStore = assign({}, EventEmitter.prototype, {
...
+      },
+      url: "/api/schema/" + schemaName + "/" + token,
+      data: formData,
+      processData: false,
+      contentType: false,
+      success: function(data) {
+        console.log(data);
+      },
+      error: function() {
+        console.log("error");
```
Comment (dduugg) [inline client/js/store/app_store.js]: use `console.error`

File: client/js/store/app_store.js [hunk 3]
```diff
@@ -24,6 +25,46 @@ var AppStore = assign({}, EventEmitter.prototype, {
...
+      beforeSend: function(request) {
+        request.setRequestHeader("x-jwt", localStorage.getItem("token"));
+      },
+      success: function(data) {
+        console.log("GET request for schema data successful.", data);
+        _displaySchema = data;
+        AppStore.emitChange();
+      },
+      error: function() {
+        console.log("GET request for schema data failed.");
```
Comment (dduugg) [inline client/js/store/app_store.js]: `console.error`

File: client/js/components/data_page.jsx [hunk 4]
```diff
@@ -60,33 +66,37 @@ var TableHeadersRow = React.createClass({
...
-
-   return (
-     <Table striped bordered condensed hover>
-        <TableHeadersRow headerKeys={headerKeys}/>
-        <TableRows rowData={fitBitData}/>
-     </Table>
-   );
+   
+   if(this.props.displaySchema){
+     var rows = this.props.displaySchema;
```
Comment (dduugg) [inline client/js/components/data_page.jsx]: do this before the `if` statement. also, do you need to check if `this.props` exists?

File: client/js/components/data_page.jsx [hunk 5]
```diff
@@ -60,33 +66,37 @@ var TableHeadersRow = React.createClass({
...
             })
           }
+          <th>Download</th>
         </tr>
       );
     }
 });
 
 var Table = React.createClass({
+ componentDidMount: function(){
```
Comment (dduugg) [inline client/js/components/data_page.jsx]: indent at 2

File: client/js/components/data_page.jsx [hunk 6]
```diff
@@ -1,14 +1,17 @@
...
 var reactBootStrap = require("react-bootstrap");
+var AppStore = require("../store/app_store.js");
 
 var UploadButton = React.createClass({
-  fileHandler: function(event) {
-    console.log(this.files);
-  },
   componentDidMount: function() {
+    var schemaName = this.props.schemaName;
+    function fileHandler(event){
```
Comment (dduugg) [inline client/js/components/data_page.jsx]: `event` is unused, consider removing. the missing space in `){` violates our style guide.

File: client/js/components/data_page.jsx [hunk 7]
```diff
@@ -60,33 +66,37 @@ var TableHeadersRow = React.createClass({
...
+      <div>
+       <Table striped bordered condensed hover>
+          <TableHeadersRow headerKeys={headerKeys}/>
+          <TableRows schemaName={this.props.schemaName} rowData={rows}/>
+       </Table>
+       <UploadButton schemaName={this.props.schemaName}/>
+      </div>
+     ); 
+   } else {
+    return <div/>;
```
Comment (dduugg) [inline client/js/components/data_page.jsx]: indent at 2

File: client/js/components/data_page.jsx [hunk 8]
```diff
@@ -60,33 +66,37 @@ var TableHeadersRow = React.createClass({
...
-   var headerKeys = Object.keys(fitBitData[0]);
-
-   return (
-     <Table striped bordered condensed hover>
-        <TableHeadersRow headerKeys={headerKeys}/>
-        <TableRows rowData={fitBitData}/>
-     </Table>
-   );
+   
+   if(this.props.displaySchema){
```
Comment (dduugg) [inline client/js/components/data_page.jsx]: whitespace issues

File: client/js/components/data_page.jsx [hunk 9]
```diff
@@ -22,72 +26,75 @@ var UploadButton = React.createClass({
...
-      );
-    }
+  render: function() {
+    var schemaName = this.props.schemaName;
+    return (
+      <tbody>
+        {
+          _.map(this.props.rowData, function(row) {
+            return (<tr>
+            {
```
Comment (dduugg) [inline client/js/components/data_page.jsx]: indentation

File: client/js/components/data_page.jsx [hunk 10]
```diff
@@ -22,72 +26,75 @@ var UploadButton = React.createClass({
...
-          {
-            _.map(this.props.headerKeys, function(header) {
-              return <th>{header}</th>;
-            })
-          }
-        </tr>
-      );
-    }
+  render: function() {
+    return (
```
Comment (dduugg) [inline client/js/components/data_page.jsx]: try to be consistent – compare line 43

File: client/js/components/data_page.jsx [hunk 11]
```diff
@@ -22,72 +26,75 @@ var UploadButton = React.createClass({
...
-     </Table>
-   );
- }
+  componentDidMount: function(){
+    AppStore.renderSchema(this.props.schemaName);
+  },
+  render: function() {
+    var Table = reactBootStrap.Table;
+    var rows = this.props.displaySchema;
+    if(rows){
```
Comment (dduugg) [inline client/js/components/data_page.jsx]: whitespace

File: client/js/components/data_page.jsx [hunk 12]
```diff
@@ -22,72 +26,75 @@ var UploadButton = React.createClass({
...
-   var headerKeys = Object.keys(fitBitData[0]);
-
-   return (
-     <Table striped bordered condensed hover>
-        <TableHeadersRow headerKeys={headerKeys}/>
-        <TableRows rowData={fitBitData}/>
-     </Table>
-   );
- }
+  componentDidMount: function(){
```
Comment (dduugg) [inline client/js/components/data_page.jsx]: whitespace

File: client/js/components/data_page.jsx [hunk 13]
```diff
@@ -22,72 +26,75 @@ var UploadButton = React.createClass({
...
+    AppStore.renderSchema(this.props.schemaName);
+  },
+  render: function() {
+    var Table = reactBootStrap.Table;
+    var rows = this.props.displaySchema;
+    if(rows){
+      var headerKeys = Object.keys(rows[0]);
```
Comment (dduugg) [inline client/js/components/data_page.jsx]: is `rows[0]` guaranteed to exist? you'll get an exception if rows is an empty array...

File: client/js/components/data_page.jsx [hunk 14]
```diff
@@ -22,72 +26,75 @@ var UploadButton = React.createClass({
...
+        <div>
+          <Table striped bordered condensed hover>
+            <TableHeadersRow headerKeys={headerKeys}/>
+            <TableRows schemaName={this.props.schemaName} rowData={rows}/>
+          </Table>
+          <UploadButton schemaName={this.props.schemaName}/>
+        </div>
+      ); 
+    } else {
+      return <div/>;
```
Comment (dduugg) [inline client/js/components/data_page.jsx]: i don't think this is valid ([source](http://stackoverflow.com/questions/7971716/is-it-ok-to-use-a-self-closing-div-tag))

File: client/js/store/app_store.js [hunk 15]
```diff
@@ -24,6 +25,40 @@ var AppStore = assign({}, EventEmitter.prototype, {
...
+        request.setRequestHeader("x-jwt", token);
+      },
+      url: "/api/schema/" + schemaName + "/" + token,
+      data: formData,
+      processData: false,
+      contentType: false,
+      success: function(data) {
+        console.log(data);
+      }
+    });
```
Comment (dduugg) [inline client/js/store/app_store.js]: no error handling?

File: client/js/store/app_store.js [hunk 16]
```diff
@@ -24,6 +25,40 @@ var AppStore = assign({}, EventEmitter.prototype, {
...
+      url: "/api/schema/" + schemaName,
+      type: "GET",
+      beforeSend: function(request) {
+        request.setRequestHeader("x-jwt", localStorage.getItem("token"));
+      },
+      success: function(data) {
+        console.log("GET request for schema data successful.", data);
+        _displaySchema = data;
+        AppStore.emitChange();
+      }
```
Comment (dduugg) [inline client/js/store/app_store.js]: no error handling?

File: client/js/components/data_page.jsx (hunk 14 above)
Comment (zindlerb) [inline client/js/components/data_page.jsx]: [react magic](http://facebook.github.io/react/tips/self-closing-tag.html)

File: client/js/components/data_page.jsx [hunk 17]
```diff
@@ -1,14 +1,18 @@
...
+var AppStore = require("../store/app_store");
 
 var UploadButton = React.createClass({
-  fileHandler: function(event) {
-    console.log(this.files);
-  },
   componentDidMount: function() {
+    var schemaName = this.props.schemaName;
+    function fileHandler() {
+      var file = this.files.item(0);
```
Comment (dduugg) [inline client/js/components/data_page.jsx]: is item(0) guaranteed to exist?

Comment (zindlerb) [inline client/js/components/data_page.jsx]: I think so. This function executes on a change event in the file upload. I think the event will only trigger if a file is uploaded.

**Comment by dduugg** [comment]
lgtm, :+1:

**Pull request merged**